frontend/packages/integration-tests-cypress/tests/crud/add-storage-crud.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { testName, checkErrors } from '../../support';
+import { listPage } from '../../views/list-page';
+import { detailsPage } from '../../views/details-page';
+import { errorMessage, submitButton } from '../../views/form';
+
+const k8sWorkloads = [
+  'replicationcontrollers',
+  'daemonsets',
+  'deployments',
+  'replicasets',
+  'statefulsets',
+];
+const openshiftWorkloads = ['deploymentconfigs'];
+const resourceObjs =
+  Cypress.env('openshift') === true ? k8sWorkloads.concat(openshiftWorkloads) : k8sWorkloads;
+
+describe('Add storage is applicable for all workloads', () => {
+  before(() => {
+    cy.login();
+    cy.createProject(testName);
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    resourceObjs.forEach((resourceType) => {
+      try {
+        cy.exec(`kubectl delete --cascade ${resourceType} example -n ${testName}`, {
+          failOnNonZeroExit: false,
+        });
+      } catch (error) {
+        console.error(`Failed to delete ${resourceType} example:\n${error}`);
+      }
+    });
+    cy.deleteProject(testName);
+    cy.logout();
+  });
+
+  resourceObjs.forEach((resourceType) => {
+    const pvcName = `${resourceType}-pvc`;
+    const pvcSize = '1';
+    const mountPath = '/data';
+    describe(resourceType, () => {
+      it(`create a ${resourceType} resource and adds storage to it`, () => {
+        listPage.createNamespacedResourceWithDefaultYAML(resourceType, testName);
+        cy.get(errorMessage).should('not.exist');
+
+        detailsPage.clickPageActionFromDropdown('Add storage');
+        cy.contains('Select claim').should('be.visible');
```

Comment:
@dtaylor113 do we ever intend to run E2E with translation enabled? I'm just curious why we are checking/using strings over IDs, or HTML `data-` attributes.

Reply:
In general our e2e tests are run in default us English.   We do run a `pseudo-translation` [test suite](https://github.com/openshift/console/blob/1867306615a027a309ce101d5878b61cfb81b660/frontend/packages/integration-tests-cypress/tests/i18n/pseudolocalization.spec.ts#L4), as well as execute multiple [cy.testI18n(...) calls](https://github.com/openshift/console/blob/f1d9bea02c3ed94fb0f6fcb9a21a67dd41728b4e/frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.spec.ts#L160) for 27 resource we run thru for our k8.crud testing -this tests various areas of the list view and detail pages.

Reply:
irt. `cy.contains('Select claim')`, yeah that is probably better served as an id, so I added a `data-test` attribute and switched to:

`cy.byTestID('claim-name').should('be.visible');`
- thanks